shimmingtoolbox/optimizer/basic_lsq.py

```diff
@@ -46,14 +44,17 @@ def optimize(self, unshimmed, mask, mask_origin=(0, 0, 0), bounds=None):
 
         mx, my, mz = mask_origin
         mX, mY, mZ = mask.shape
+        mV = mX * mY * mZ
+        mask_vec = mask.reshape((mV,))
 
-        # Set up masked submatrices
-        masked_unshimmed = np.where(mask != 0, unshimmed[mx:mx+mX, my:my+mY, mz:mz+mZ], 0)
-        masked_coils = np.where(mask.reshape(mask.shape + (1,)) != 0, self.coils[mx:mx+mX, my:my+mY, mz:mz+mZ, :], 0)
+        # Set up optimization vectors
+        coil_mat = np.reshape(np.transpose(self.coils[mx:mx+mX, my:my+mY, mz:mz+mZ], axes=(3, 0, 1, 2)), (self.N, mV)).T[mask_vec != 0, :] # mV' x N
```

Comment:
i'm a bit confused as to what "mask_origin" corresponds to? could you give an example?